src/lib/utils/validation/validation.md

```diff
@@ -1,3 +1,155 @@
+  validateComponents,
+} = Utils.Validation;
+const {
+  Input,
+  Select,
+  TextBox,
+  RadioButton,
+} = Components;
+
+this.state = {
+  universityEmail: '',
+  firstName: '',
+  hasErrorAfterSubmit: false,
+}
+
+this.validationObject = {
+  firstName: {
+    validations: [maxCharacters(300), required()],
+  },
+  universityEmail: {
+    validations: [maxCharacters(300), validateEmail()],
+  }
+}
+
+submit = () => {
+  const validateObject = validateComponents(this.validationsObject, this.state);
+  if (validateObject.hasError) {
+    this.setState({ hasErrorAfterSubmit: true })
+    return;
+  }
+  submitData(this.state);
+}
+// .... code
+// .... more code
+// .... Render Method
+  <Input
+    containerClassName={styles.inputWrapper}
+    value={firstName}
+    placeholder="First Name"
+    onChangeFunction={this.updateState}
+    name="firstName"
+    validations={this.validationObject[firstName]}
+    hasErrorAfterSubmit={hasErrorAfterSubmit}
+    autoFocus
+  />
+  <Input
+    containerClassName={styles.inputWrapper}
+    value={universityEmail}
+    placeholder="University/College email"
+    onChangeFunction={this.updateState}
+    name="universityEmail"
+    type="email"
+    validations={this.validationObject[universityEmail]}
+    hasErrorAfterSubmit={hasErrorAfterSubmit}
+  />
+  <button onClick={this.submit}
+```
+Take into account that in this case we use an object `validationObject` where we save every field with the name of the key as the name of the component. After that, when the user submit the data we re-evaluate all the fields calling to `validateComponents` where we send the object with the validations we need to make and another object (in this case the state) with the values of the fields (this was made like that because usually we have all the values of the fields in the state or in the props of the components).
+
+When the validations after the submition failed, we set in the state a variable `hasErrorAfterSubmit` to true and we send that as parameter to one of our component. In this case you are forcing every component to validate their selves (without having to type in the fields).
+
+## Types of Validations
+The validations available are:
+- required
+- validateNumeric
+- validateAlphanumeric
+- validateEmail
+- validateNonNegative
+- validateHigherThanZero
+- minAmount
+- maxAmount
+- maxCharacters
+- minCharacters
+- validDate
+
+The min and max validation receive as param the min or max value that you want to evaluate. For example, maxCharacters(300) receive the number 300 to know that this is the maximum amount of characters. 
+All validation receive the custom message (if they have another param, it would be the second) and this is optional.
+
+
+## The validations object
+This object should have the `validations` field and it could have another field call `condition` for every field we want to validate. 
+The field `condition` is a condition to evaluate if you want to execute that validation or not. 
```

Comment:
the field condition is a condition sound like repetitive...we can say The field condition is used to check if we it should execute the validation or it's not necessary...

Reply:
👍